### DISCUSSION TITLE
How to migrate ~250 shares from old AIO instance to new one?

### DISCUSSION THREAD

**Discussion by BraidenPsiuk** [nextcloud/all-in-one#4470] [Questions] [answered]
# How can I migrate 250 share links/tokens from one AIO instance to another?

I tried searching for an answer already, but haven't found any good guides specific to AIO.

## My Scenario
I am getting ready to migrate an AIO instance to a new server. On the old server, I had created a few hundred (~250) public share links (all under the same user). Ideally, I would just copy over my borgbackup directory, restore the most recent backup, and be done.

But, my old server uses AIO in "normal" mode, while I want to use AIO in "reverse proxy" mode on the new server. As far as I can tell, if I do a restore from backup, I can't change from one 'mode' to the other (although I haven't tried this- please tell me if I'm wrong!).

So I've considered doing the migration just by copying stuff over- I can copy all files over to the new instance, that's fairly simple. But as far as all of my public share links, how might I move that information over? The way my shares are done is very predictable, all shared objects are directories, and they're all sub-directories which reside under the same parent directory, like this:

```
Top level directory /
  - "Folder 1" (publicly shared via link "https://example.com/s/A...")
  - "Folder 2" (publicly shared via link "https://example.com/s/B...)")
  - ...etc, approximately 250 of these
```

Maybe I'm approaching this from the wrong angle and there is a simpler way to do the migration. I tried the `User Migration` app, but that doesn't specifically mention anything about supporting migration of shares. If it does though, that might work. I also tried using the API to loop through and grab the share tokens which seems do-able, but I'm not so sure how I would apply them to the new instance once I have built up a list of them and the directories they belong to. I'd also prefer to avoid this sort of scripting if possible. (But if I do go the API/scripting route, I'll gladly share the script I create)

Any direction here would be greatly appreciated!

**Comment by szaimen** [comment, marked as answer]
> Ideally, I would just copy over my borgbackup directory, restore the most recent backup, and be done.

Yes this should be possible.



> But, my old server uses AIO in "normal" mode, while I want to use AIO in "reverse proxy" mode on the new server. As far as I can tell, if I do a restore from backup, I can't change from one 'mode' to the other (although I haven't tried this- please tell me if I'm wrong!).

Actually you are wrong. If you follow https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md, it will keep the reverse proxy mode even if you restore your instance from backup.

**Comment by BraidenPsiuk** [comment]
Thank you! [Selected answer](https://github.com/nextcloud/all-in-one/discussions/4470#discussioncomment-8967371) worked well and you're right, it still used reverse proxy mode.

The migration I mentioned went very well. Only ran into two slight hiccups so I'll document them here in case someone else comes across them also:
1. In the new instance: I was stuck in a state where I had previously attempted a backup without my backup volume mounted so the backup container failed. And I wanted to do another restore, but I no longer had the option to restore from backup (without first doing another successful backup, but I didn't want to do this since it would take a little while to complete). I tried `docker exec nextcloud-aio-mastercontainer rm /mnt/docker-aio-config/data/borg.config` but I think this helps more when you are trying to backup, not restore. **Simple solution for me was to just [reset the instance](https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-properly-reset-the-instance) and then follow the recovery process again.**
2. The `Open Nextcloud AIO Interface` button in `Administration > Overview` kept linking back to the old server's IP. **The solution was [surprisingly simple](https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-easily-log-in-to-the-aio-interface) :)**
